test/integration/storageversion/storage_version_filter_test.go

```diff
@@ -0,0 +1,235 @@
+	assertBlocking("writes to Storage Version", t, err, shouldBlock)
+}
+
+func testNonPersistedWrite(t *testing.T, cfg *rest.Config, shouldBlock bool) {
+	client := clientset.NewForConfigOrDie(cfg)
+	_, err := client.AuthenticationV1().TokenReviews().Create(context.TODO(), &authenticationv1.TokenReview{
+		Spec: authenticationv1.TokenReviewSpec{
+			Token: "some token",
+		},
+	}, metav1.CreateOptions{})
+	assertBlocking("non-persisted write", t, err, shouldBlock)
+}
+
+func testBuiltinResourceRead(t *testing.T, cfg *rest.Config, shouldBlock bool) {
+	client := clientset.NewForConfigOrDie(cfg)
+	_, err := client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	assertBlocking("reads of built-in resources", t, err, shouldBlock)
+}
+
+// TestStorageVersionBootstrap ensures that before the StorageVersions are
+// updated, only the following the request are accepted by the apiserver:
+// 1. read requests
+// 2. non-persisting write requests
+// 3. write requests to the storageversion API
+// 4. requests to CR or aggregated API
+func TestStorageVersionBootstrap(t *testing.T) {
+	// Start server and create CRD
+	etcdConfig := framework.SharedEtcd()
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, nil, etcdConfig)
+	etcd.CreateTestCRDs(t, apiextensionsclientset.NewForConfigOrDie(server.ClientConfig), false, etcd.GetCustomResourceDefinitionData()[0])
+	server.TearDownFn()
+
+	startUpdateSV := make(chan struct{})
+	finishUpdateSV := make(chan struct{})
+	updateFinished := make(chan struct{})
+	completed := make(chan struct{})
+	wrapperFunc := func(delegate storageversion.Manager) storageversion.Manager {
+		return &wrappedStorageVersionManager{
+			startUpdateSV:  startUpdateSV,
+			finishUpdateSV: finishUpdateSV,
+			updateFinished: updateFinished,
+			completed:      completed,
+			Manager:        delegate,
+		}
+	}
+	// Restart api server, enable the storage version API and the feature gate.
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StorageVersionAPI, true)()
+	server = kubeapiservertesting.StartTestServerOrDie(t,
+		&kubeapiservertesting.TestServerInstanceOptions{
+			StorageVersionWrapFunc: wrapperFunc,
+		},
+		[]string{
+			// force enable all resources to ensure that the storage updates can handle cross group resources.
+			// TODO: drop these once we stop allowing them to be served.
+			"--runtime-config=api/all=true,extensions/v1beta1/deployments=true,extensions/v1beta1/daemonsets=true,extensions/v1beta1/replicasets=true,extensions/v1beta1/podsecuritypolicies=true,extensions/v1beta1/networkpolicies=true,internal.apiserver.k8s.io/v1alpha1=true",
+		},
+		etcdConfig)
+	defer server.TearDownFn()
+
+	cfg := rest.CopyConfig(server.ClientConfig)
+
+	// Write to k8s native API object should be blocked.
+	testBuiltinResourceWrite(t, cfg, true)
```

Comment:
Consider using t.Run() for these sub-tests?

Reply:
ack

Reply:
added
```
--- PASS: TestStorageVersionBootstrap (7.97s)
    --- PASS: TestStorageVersionBootstrap/before_storage_version_update (1.72s)
        --- PASS: TestStorageVersionBootstrap/before_storage_version_update/write_to_k8s_native_API_object_should_be_blocked (1.68s)
        --- PASS: TestStorageVersionBootstrap/before_storage_version_update/write_to_CRD_should_be_blocked (0.00s)
        --- PASS: TestStorageVersionBootstrap/before_storage_version_update/write_to_APIService_should_be_blocked (0.00s)
        --- PASS: TestStorageVersionBootstrap/before_storage_version_update/write_to_CR_should_pass (0.03s)
        --- PASS: TestStorageVersionBootstrap/before_storage_version_update/write_to_the_storage_version_API_should_pass (0.00s)
        --- PASS: TestStorageVersionBootstrap/before_storage_version_update/write_to_non-persisted_API_should_pass (0.00s)
        --- PASS: TestStorageVersionBootstrap/before_storage_version_update/read_of_k8s_native_API_should_pass (0.00s)
    --- PASS: TestStorageVersionBootstrap/after_storage_version_update (0.02s)
        --- PASS: TestStorageVersionBootstrap/after_storage_version_update/write_to_k8s_native_API_object_should_pass (0.00s)
        --- PASS: TestStorageVersionBootstrap/after_storage_version_update/write_to_CRD_should_pass (0.00s)
        --- PASS: TestStorageVersionBootstrap/after_storage_version_update/write_to_APIService_should_pass (0.00s)
        --- PASS: TestStorageVersionBootstrap/after_storage_version_update/write_to_the_storage_version_API_should_pass (0.01s)
        --- PASS: TestStorageVersionBootstrap/after_storage_version_update/write_to_non-persisted_API_should_pass (0.00s)
        --- PASS: TestStorageVersionBootstrap/after_storage_version_update/read_of_k8s_native_API_should_pass (0.00s)
    --- PASS: TestStorageVersionBootstrap/after_storage_version_manager_complete (0.60s)
PASS
ok  	k8s.io/kubernetes/test/integration/storageversion	8.324s
```